pgxn/neon/control_plane_connector.c

```diff
@@ -433,32 +433,30 @@ MergeTable()
 											   entry->name,
 											   HASH_ENTER,
 											   NULL);
+			RoleEntry * old;
+			bool found_old = false;
 
 			to_write->type = entry->type;
-			if (entry->password)
-				to_write->password = entry->password;
+			to_write->password = entry->password;
 			strlcpy(to_write->old_name, entry->old_name, NAMEDATALEN);
-			if (entry->old_name[0] != '\0')
-			{
-				bool		found_old = false;
-				RoleEntry  *old = hash_search(
-											  CurrentDdlTable->role_table,
-											  entry->old_name,
-											  HASH_FIND,
-											  &found_old);
-
-				if (found_old)
-				{
-					if (old->old_name[0] != '\0')
-						strlcpy(to_write->old_name, old->old_name, NAMEDATALEN);
-					else
-						strlcpy(to_write->old_name, entry->old_name, NAMEDATALEN);
-					hash_search(CurrentDdlTable->role_table,
-								entry->old_name,
-								HASH_REMOVE,
-								NULL);
-				}
-			}
+			if (entry->old_name[0] == '\0')
+				continue;
```

Comment:
There doesn't seem to be any point in doing the `hash_search()` above, the `strlcpy()`, or the `type` and `password` assignments if we hit this early continue.

Reply:
No, we need to insert the item, otherwise the drop is not processed correctly and this test fails:
```
    cur.execute("CREATE ROLE bork WITH PASSWORD 'newyork'")
    cur.execute("BEGIN")
    cur.execute("SAVEPOINT point")
    cur.execute("DROP ROLE bork")
    cur.execute("COMMIT")
```